proposals/lip-Introduce_PoA_module.md

```diff
@@ -0,0 +1,606 @@
+* Each store value is the serialization of an object following `snapshotStoreSchema`.
+* Notation: Let `snapshotStore(roundNumber)` be the store entry with prefix `STORE_PREFIX_SNAPSHOT` and key `uint32be(roundNumber)`.
+
+##### JSON Schema
+
+```java
+snapshotStoreSchema = {
+    "type": "object",
+    "properties": {
+        "validators": {
+            "type": "array",
+            "fieldNumber": 1,
+            "items": {
+                "type": "object",
+                "required": ["address", "weight"],
+                "properties": {
+                    "address": {
+                        "dataType": "bytes",
+                        "fieldNumber": 1
+                    },
+                    "weight": {
+                        "dataType": "uint64",
+                        "fieldNumber": 2
+                    }
+                }
+            }
+        },
+        "threshold": {
+            "dataType": "uint64",
+            "fieldNumber": 2
+        }
+    },
+    "required": ["validators", "threshold"]
+}
+ ```
+
+##### Properties and Default Values
+
+The properties of this schema are as follows:
+
+  * Each element in the `validators` array corresponds to a validator and stores its address and weight property. The elements in the array must be sorted lexicographically by `address` property.
+It specifies the set of active validators in the chain. 
+Its initial value is set in the genesis block.
+  * `threshold`: An integer stating the weight threshold for finality in the BFT consensus protocol.
+Its initial value is set in the genesis block.
+
+#### Chain Properties Substore
+
+This substore contains the general properties of the chain.
+
+##### Store Prefix, Store Key, and Store Value
+
+* The store prefix is set to `STORE_PREFIX_CHAIN`.
+* The store key is set to empty bytes.
+* The store value is set to the serialization of an object following `chainPropSchema` below.
+* Notation: Let `chainProperties` be the entry in the chain properties substore.
+
+##### JSON Schema
+
+```java
+chainPropSchema = {
+    "type": "object",
+    "properties": {
+        "roundEndHeight": {
+            "dataType": "uint32",
+            "fieldNumber": 1
+        },
+        "validatorsUpdateNonce": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    },
+    "required": ["roundEndHeight", "validatorsUpdateNonce"]
+}
+```
+
+##### Properties and Default Values
+
+The properties of this schema are as follows:
+
+*   `roundEndHeight`: An integer stating the last height of the round. 
+Its initial value is set after the execution of the genesis block.
+*   `validatorsUpdateNonce`: An integer representing the number of times that the validator set has been updated with an update authority command. 
+It is initialized to 0.
+
+### Commands
+
+#### Authority Registration Command
```

Comment:
Should we consider having a fee for this registration similar to DPoS with the same rational?

Reply:
good point, added for completeness.